Random fail of test using `scipy.curve_fit`

Unrelated failure during build test in pull request:  https://github.com/scipp/scipp/pull/2929


```
=================================== FAILURES ===================================
2023-01-06T13:27:28.9961250Z ___ test_optimized_params_approach_real_params_as_data_noise_decreases[0.1] ____
2023-01-06T13:27:28.9962740Z [gw2] darwin -- Python 3.8.15 /Users/runner/work/scipp/scipp/.tox/test/bin/python
2023-01-06T13:27:28.9963150Z 
2023-01-06T13:27:28.9963350Z noise_scale = 0.1
2023-01-06T13:27:28.9963630Z 
2023-01-06T13:27:28.9964270Z     @pytest.mark.parametrize("noise_scale", [1e-1, 1e-2, 1e-3, 1e-6, 1e-9])
2023-01-06T13:27:28.9964670Z     def test_optimized_params_approach_real_params_as_data_noise_decreases(noise_scale):
2023-01-06T13:27:28.9965090Z         popt, _ = curve_fit(func, array1d(a=1.7, b=1.5, noise_scale=noise_scale))
2023-01-06T13:27:28.9965610Z         assert sc.allclose(popt['a'], sc.scalar(1.7), rtol=sc.scalar(2.0 * noise_scale))
2023-01-06T13:27:28.9966110Z >       assert sc.allclose(popt['b'], sc.scalar(1.5), rtol=sc.scalar(2.0 * noise_scale))
2023-01-06T13:27:28.9966410Z E       assert False
2023-01-06T13:27:28.9966890Z E        +  where False = <function allclose at 0x7fa0f2d31280>(<scipp.Variable> ()    float64  [dimensionless]  [1.8065]  [0.00740935], <scipp.Variable> ()    float64  [dimensionless]  [1.5], rtol=<scipp.Variable> ()    float64  [dimensionless]  [0.2])
2023-01-06T13:27:28.9967410Z E        +    where <function allclose at 0x7fa0f2d31280> = sc.allclose
2023-01-06T13:27:28.9967790Z E        +    and   <scipp.Variable> ()    float64  [dimensionless]  [1.5] = <function scalar at 0x7fa0f2bc7f70>(1.5)
2023-01-06T13:27:28.9968160Z E        +      where <function scalar at 0x7fa0f2bc7f70> = sc.scalar
2023-01-06T13:27:28.9969020Z E        +    and   <scipp.Variable> ()    float64  [dimensionless]  [0.2] = <function scalar at 0x7fa0f2bc7f70>((2.0 * 0.1))
2023-01-06T13:27:28.9969470Z E        +      where <function scalar at 0x7fa0f2bc7f70> = sc.scalar
2023-01-06T13:27:28.9969650Z 
2023-01-06T13:27:28.9969800Z tests/scipy/optimize/curve_fit_test.py:120: AssertionError
2023-01-06T13:27:28.9970120Z =========================== short test summary info ============================
2023-01-06T13:27:28.9970470Z SKIPPED [1] tests/datetime_test.py:265: Requires std::chrono from C++20
2023-01-06T13:27:28.9970830Z SKIPPED [1] tests/bins_test.py:417: Need fix in Variable::setSlice
2023-01-06T13:27:28.9971160Z SKIPPED [14] tests/constants/constants_test.py:14: Skipped
2023-01-06T13:27:28.9971720Z SKIPPED [1] tests/plotting/plot_2d_test.py:463: Require in-place concatenate
2023-01-06T13:27:28.9972160Z FAILED tests/scipy/optimize/curve_fit_test.py::test_optimized_params_approach_real_params_as_data_noise_decreases[0.1]
2023-01-06T13:27:28.9972570Z ============ 1 failed, 3950 passed, 17 skipped in 141.20s (0:02:21) ============
```

--- a/tests/scipy/optimize/curve_fit_test.py
+++ b/tests/scipy/optimize/curve_fit_test.py
@@ -18,7 +18,7 @@ def array1d(*, a=1.2, b=1.3, noise_scale=0.1, size=50):
     y = func(x, a=a, b=b)
     rng = np.random.default_rng()
     # Noise is random but avoiding unbounded values to avoid flaky tests
-    y.values += noise_scale * np.clip(rng.normal(size=size), -2.0, 2.0)
+    y.values += noise_scale * np.clip(rng.normal(size=size), -1.5, 1.5)
     return sc.DataArray(y, coords={'xx': x})
 
 

